pymongo/common.py

```diff
@@ -856,15 +864,14 @@ def get_setter_key(x: str) -> str:
     for opt, value in options.items():
         normed_key = get_normed_key(opt)
         try:
-            validator = URI_OPTIONS_VALIDATOR_MAP.get(normed_key, raise_config_error)
```

Comment:
This code should still be using URI_OPTIONS_VALIDATOR_MAP but by calling validate() it now uses VALIDATORS. This is why I suggested the `_get_validator` function above.